woodwork/tests/accessor/test_table_accessor.py

```diff
@@ -2117,6 +2118,28 @@ def test_setitem_invalid_input(sample_df):
         df.ww['signup_date'] = df.signup_date
 
 
+def test_setitem_indexed_column(sample_df):
+    sample_df_with_index = sample_df
+    sample_df_with_index.ww.init()
+    sample_df.ww.init()
+    sample_df_with_index.ww.set_index('id')
+    inputs = [sample_df, sample_df_with_index]
+
+    # Testing two cases: The dataframe has no index, The dataframe already had an index that was removed
+    for input_ in inputs:
```

Comment:
lets split the two cases into separate tests. I'd say it's only worth doing this kind of format if the code is exactly the same from input to input. But once we start having to add if statements for each input, we should put them in separate tests. (or, if the tests still seem similar enough to be in one test function, just put one after another)

In general, I'd stick with separate tests unless you realize you're about to copy and paste a bunch of code.